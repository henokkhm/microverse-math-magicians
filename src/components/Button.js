import PropTypes from 'prop-types';

import styles from '../styles/Button/Button.module.css';

function Button({
  value, wide, accent, handleClick,
}) {
  const buttonClasses = `${styles.button} ${wide ? styles.wide : ''} ${
    accent ? styles.accent : ''
  }`;
  return (
    <button
      className={buttonClasses}
      type="button"
      onClick={() => handleClick(value)}
    >
      {value}
    </button>
  );
}

Button.propTypes = {
  value: PropTypes.string.isRequired,
  wide: PropTypes.bool,
  accent: PropTypes.bool,
  handleClick: PropTypes.func.isRequired,
};

Button.defaultProps = {
  wide: false,
  accent: false,
};

export default Button;
