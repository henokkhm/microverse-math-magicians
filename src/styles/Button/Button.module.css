.button {
  font-size: 1.1rem;
  width: 5rem;
  height: 5rem;
  padding: 2rem;
  border: none;
  outline: none;
  background: #e0e0e0;
  transition: all 100ms ease-in-out;
  display: flex;
  justify-content: center;
  align-items: center;
}

.button:hover {
  background: #cdcaca;
}

.button:active {
  transform: translateY(2px) scale(0.99);
}

.wide {
  width: calc(10rem + 1px);
}

.accent {
  background: #f5913e;
}

.accent:hover {
  background: #d87e35;
}
