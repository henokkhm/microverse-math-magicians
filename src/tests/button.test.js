import renderer from 'react-test-renderer';
import Button from '../components/Button';

describe('The button', () => {
  it('renders correct value', () => {
    const tree = renderer.create(<Button value="hi" handleClick={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correct value', () => {
    const tree = renderer.create(<Button value="hello" handleClick={() => {}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correct className when we pass wide props', () => {
    const tree = renderer.create(<Button value="good" handleClick={() => {}} wide />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correct className when we pass accent props', () => {
    const tree = renderer.create(<Button value="good" handleClick={() => {}} accent />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
