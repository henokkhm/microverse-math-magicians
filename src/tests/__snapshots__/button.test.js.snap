// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Navbar renders correctly 1`] = `
<button
  className="button  "
  onClick={[Function]}
  type="button"
>
  hi
</button>
`;

exports[`The button renders correct className when we pass accent props 1`] = `
<button
  className="button  accent"
  onClick={[Function]}
  type="button"
>
  good
</button>
`;

exports[`The button renders correct className when we pass wide props 1`] = `
<button
  className="button wide "
  onClick={[Function]}
  type="button"
>
  good
</button>
`;

exports[`The button renders correct css when we pass wide props 1`] = `
<button
  className="button wide "
  onClick={[Function]}
  type="button"
>
  good
</button>
`;

exports[`The button renders correct value 1`] = `
<button
  className="button  "
  onClick={[Function]}
  type="button"
>
  hi
</button>
`;

exports[`The button renders correct value 2`] = `
<button
  className="button  "
  onClick={[Function]}
  type="button"
>
  hello
</button>
`;
